examples/policies-with-custom-data-sources/README.md

```diff
@@ -0,0 +1,213 @@
+# Policies with custom data sources
+
+This example creates a policy via terraforms [`aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) and then adds it to a role's `policy_arns` list.
+
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Example](#example)
+    - [Variable definition](#variable-definition)
+    - [Use data source to fetch a dynamic value](#use-data-source-to-fetch-a-dynamic-value)
+    - [Build our policy](#build-our-policy)
+    - [Enrich roles list with created policy](#enrich-roles-list-with-created-policy)
+    - [Define the iam module](#define-the-iam-module)
+5. [Usage](#usage)
+6. [Requirements](#requirements)
+7. [Providers](#providers)
+8. [Inputs](#inputs)
+9. [Outputs](#outputs)
+
+
+## Overview
+
+By using [`aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) you have the advantage to use terraform's variables inside the policy definition and be able to first gather already existing resources and add them to that policy. This makes the process of creating policies as dynamic and flexible as it can get.
```

Comment:
Or, you can just do [this](https://github.com/Flaconi/terraform-aws-iam-roles/pull/7/files/844fce50e1ae053b8984a37111886992eaa71b91..fb56ca0b71084f6d26bd7d938289bc8a2a496e3c#diff-b2c85bb03aef5106d9c66669dd75d6298880470f3ead6da1c86f451dc2399df5R23) 😃

Reply:
Indeed, if that is sufficient for you. The example shows an extra functionality. Sometimes you need to gather data first that you are not aware of.